src/vivarium/framework/utilities.py

```diff
@@ -28,20 +29,91 @@ def to_yearly(value: TimeValue, time_step: Timedelta) -> TimeValue:
     return value / (time_step.total_seconds() / (60 * 60 * 24 * 365.0))
 
 
-def rate_to_probability(rate: Sequence[float] | NumberLike) -> NumericArray:
-    # encountered underflow from rate > 30k
-    # for rates greater than 250, exp(-rate) evaluates to 1e-109
-    # beware machine-specific floating point issues
+def rate_to_probability(
+    rate: Sequence[float] | NumberLike,
+    time_scaling_factor: float | int = 1.0,
+    rate_conversion_type: Literal["linear", "exponential"] = "linear",
+) -> NumericArray:
+    """Converts a rate to a probability.
+
+    Parameters
+    ----------
+    rate
+        The rate to convert to a probability.
+    time_scaling_factor
+        The time factor in to scale the rates by. This is usually the time step.
+    conversion_type
+        The type of conversion to use. Default is "linear" for a simple multiplcation
+        of rate and time_scaling_factor. The other option is "exponential" which should be
+        used for continuous time event driven models.
+
+    Returns
+    -------
+        An array of floats representing the probability of the converted rates
+    """
+    if rate_conversion_type not in ["linear", "exponential"]:
```

Comment:
Is there a reason to have this check both here and in randomness.stream.py?

Reply:
Yes because this is a utility function so if it called outside of the randomness stream we still want the check to happen.